Portal/src/Datahub.Portal/Pages/Help/HelpPage.razor

```diff
@@ -420,48 +419,31 @@
-
-        using (var client = new HttpClient())
+        var query = $"SELECT [System.Id], [System.Title], [System.CreatedBy], [System.CreatedDate], [System.ChangedDate], [System.Tags] FROM WorkItems WHERE [Submitted By] = '{userName}' AND NOT [System.Title] CONTAINS 'xception'";
+        var wiql = new Wiql()
         {
-            var credentials = await _keyVaultService.GetSecret(_portalConfiguration.AdoServiceUser.PatSecretName);
-            client.DefaultRequestHeaders.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));
-            client.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Basic", Convert.ToBase64String(System.Text.Encoding.ASCII.GetBytes($":{credentials}")));
-
-            // Create request body and send the POST request to Azure DevOps API
-            var body = new JObject { ["query"] = query };
-            var response = await client.PostAsync(apiUrl, new StringContent(body.ToString(), System.Text.Encoding.UTF8, "application/json"));
-            response.EnsureSuccessStatusCode();
-
-            // Parse the response and extract the work item titles
-            var jsonResponse = await response.Content.ReadAsStringAsync();
-            var result = JObject.Parse(jsonResponse);
-
-            // Collect all the titles
-            var workItems = (JArray)result["workItems"];
+            Query = query
+        };
 
-            foreach (var workItem in workItems)
-            {
-                // Get the WorkItem ID
-                var workItemId = (int)workItem["id"];
-                // Get the WorkItem details with ID
-                var itemApiUrl = $"https://dev.azure.com/{_portalConfiguration.AdoOrg.OrgName}/{_portalConfiguration.AdoOrg.ProjectName}/_apis/wit/workItems/{workItemId}?api-version=7.0";
-                var workItemResponse = await client.GetAsync(itemApiUrl);
-                workItemResponse.EnsureSuccessStatusCode();
+        // Configure the Azure DevOps client
+        var config = new AzureDevOpsConfiguration();
+        config.TenantId = _portalConfiguration.AzureAd.TenantId;
+        config.ClientId = _portalConfiguration.AzureAd.InfraClientId;
+        config.ClientSecret = _portalConfiguration.AzureAd.InfraClientSecret;
+        var clientProvider = new AzureDevOpsClient(config);
 
-                var workItemJsonResponse = await workItemResponse.Content.ReadAsStringAsync();
-                var workItemDetails = JObject.Parse(workItemJsonResponse);
+        var client2 = await clientProvider.WorkItemClientAsync();
+        var res = await client2.QueryByWiqlAsync(wiql).ConfigureAwait(false);
+        var ids = res.WorkItems.Select(item => item.Id).ToArray();
 
-                // Add the issue to the list
-                var issue = new IssueForDisplaying(workItemDetails, true);
+        foreach (var id in ids)
```

Comment:
Is there a reason why you dont use `client2.GetWorkItemsAsync(ids)`? This would make a single call instead of 1 call/issue, and you could keep the for loop for getting the IssueForDisplaying

Reply:
Updated to use that instead